### ISSUE TITLE
Maximum number of 3rd Party Cameras

### ISSUE THREAD

**Issue by thomason-jesse** [allenai/ai2thor#510]
It `AddThirdPartyCamera` fails for N + 1 for `agentCount=N`. Is this a hard constraint? It would be nice to be able to have more than one third party camera per agent.

```
from ai2thor.controller import Controller
controller = Controller(gridSize=0.25, agentCount=1, snapToGrid=True, renderObjectImage=True, visibilityDistance=1.5)
event = controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=0, y=0, z=90), position=dict(x=-1.0, z=-2.0, y=1.0)))
event = controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=0, y=0, z=90), position=dict(x=-1.0, z=-2.0, y=1.0)))     <--- Crashes
```

```
from ai2thor.controller import Controller
controller = Controller(gridSize=0.25, agentCount=2, snapToGrid=True, renderObjectImage=True, visibilityDistance=1.5)
event = controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=0, y=0, z=90), position=dict(x=-1.0, z=-2.0, y=1.0)))
event = controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=0, y=0, z=90), position=dict(x=-1.0, z=-2.0, y=1.0)))
event = controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=0, y=0, z=90), position=dict(x=-1.0, z=-2.0, y=1.0)))     <--- Crashes
```

**Comment by thomason-jesse** [comment]
As of https://github.com/allenai/ai2thor/commit/520ab0915c6d0bc2a4cf01bf8af0bedd6145426d the hanging calls about now result in an `IndexError: list index out of range` rather than hanging.

**Comment by ekolve** [comment]
I don't get an error, but it does hang for me.  It appears to be related to the imageSynthesis plugin for objectImage.  Do you get an error with ```renderObjectImage=False```

**Comment by thomason-jesse** [comment]
Confirmed this is fixed as of 2.5.3.
